server/src/test/java/com/octopus/teamcity/opentelemetry/server/OTELHelperTest.java

```diff
@@ -46,39 +47,93 @@ void isReadyTest() {
     }
 
     @Test
-    void getParentSpanTest(@Mock SRunningBuild build) {
-        assertEquals(this.otelHelper.getParentSpan(String.valueOf(build.getBuildId())), this.otelHelper.getSpan(String.valueOf(build.getBuildId())) );
+    void getParentSpanShouldReturnABuildSpanAndBeAvailableInGetSpan(@Mock SRunningBuild build) {
+        // Arrange
+        String parentBuildId = String.valueOf(build.getBuildId());
+        Span parentSpan = this.otelHelper.getParentSpan(parentBuildId);
+
+        // Act
+        Span buildSpan = this.otelHelper.getSpan(String.valueOf(build.getBuildId()));
+
+        // Assert
+        assertNotNull(parentSpan);
+        assertEquals(parentSpan, buildSpan);
     }
 
     @Test
-    void createSpanTest(@Mock SRunningBuild build) {
-        Span parentSpan = this.otelHelper.getParentSpan(String.valueOf(build.getBuildId()));
-        assertEquals(this.otelHelper.createSpan(String.valueOf(build.getBuildId()), parentSpan), this.otelHelper.getSpan(String.valueOf(build.getBuildId())));
+    void createSpanShouldHaveSpanAvailableInGetSpan() {
+        // Arrange
+        SRunningBuild build = mock(SRunningBuild.class, RETURNS_DEEP_STUBS);
+        Span parentSpan = createParentSpanForTest(build);
+
+        // Act
+        Span expectedSpan = this.otelHelper.createSpan(String.valueOf(build.getBuildId()), parentSpan);
+        Span actualSpan = this.otelHelper.getSpan(String.valueOf(build.getBuildId()));
+
+        // Assert
+        assertEquals(expectedSpan, actualSpan);
     }
 
     @Test
-    void createTransientSpanTest(@Mock SRunningBuild build) {
-        Span parentSpan = this.otelHelper.getParentSpan(String.valueOf(build.getBuildId()));
-        assertNotNull(this.otelHelper.createTransientSpan(String.valueOf(build.getBuildId()), parentSpan, new Date().getTime()));
+    void createTransientSpanShouldReturnSpanAndNotBeAvailableInGetSpan() {
+        // Arrange
+        SRunningBuild build = mock(SRunningBuild.class, RETURNS_DEEP_STUBS);
+        SRunningBuild parentBuild = mock(SRunningBuild.class, RETURNS_DEEP_STUBS);
+        Span parentSpan = createParentSpanForTest(parentBuild);
+        // Work around for mocked build.getBuildId(), as you cannot mock a final method (will return 0 in all cases when called), this ensures that the parent build id and the build id are different values
+        // This ensures two different spans are created, a build span and a parent span
+        when(build.getBuildId()).thenReturn(12345L);
+        String buildId = String.valueOf(build.getBuildId());
+
+        // Act
+        Span expectedSpan = this.otelHelper.createTransientSpan(buildId, parentSpan, new Date().getTime());
+
+        // Assert
+        assertNotNull(expectedSpan);
+        assertNotNull(String.valueOf(parentBuild.getBuildId()));
```

Comment:
```suggestion
        assertNotNull(this.otelHelper.getSpan(parentBuild.getBuildId());
```
but with correct syntax

Reply:
done!